### DISCUSSION TITLE
Feature Improvements that zen needs!!!!

### DISCUSSION THREAD

**Discussion by nanoraptor** [zen-browser/desktop#2078] [Ideas]
### Feature improvements in zen:

- Extension options available by from the extensions menu in the homepage
- Split view indicator in tabs (This one should've been integrated when zen was even launched) [#1938](https://github.com/zen-browser/desktop/issues/1938) #2038 
- PWA Support (Mozilla stopped it. If the zen team can bring this one, it will be great!) #800 
- It should have these stuff built in: 
      
    1. [TitleBarButton UI Tweaks](https://zen-browser.app/themes/180d9426-a020-4bd7-98ec-63f957291119) 
    2. [Cleaner Extension Menu ](https://zen-browser.app/themes/1e86cf37-a127-4f24-b919-d265b5ce29a0)
    3. [Private Mode Highlighting](https://zen-browser.app/themes/58649066-2b6f-4a5b-af6d-c3d21d16fc00) 
    4. [Monochrome Color Scheme](https://zen-browser.app/mods/11d685eb-4515-4045-864b-0a50589f8a4d)
   

- Containers' style on the address bar should be changed to make it more aligned with the aesthetics of the browser itself or maybe buliding this feature right into the browser itself. Each workspaces by default will have different containers(Each having their own set of logins and stuff). Users can then use subcontainers within the workspace by using mozilla's extension
- Pressing esc to get out of address bar (seriously, I never thought something this easy has to be done by pressing f6. fix this) #1142 
- Abiliity to one-click toggle between the night mode and light mode (with the preferences of each modes being intact)
- Arc Browser Features such as:

  1. Tab grouping like arc #1579 
  2. Per Workspace bookmarks and pinned tabs along with the golbal ones (atleast an option) #885 
  3. A command pallete similar to that of arc which lets us basically do anything #820 
  4. Peek Window Feature  #2109 

- Private tab/container #767 
- Horizontal Tabs support #889 
- Windows ARM64 support #858 and linux on arm support #883 
- Support for Vim keybindings Natively #2017 
- consistent UI/UX #2073
- A better start page #1134 
- Ability to open a new tab within split view #2137 
- Ability to configure/modify shortcuts like ctrl+K which interfere with spotify's shortcuts
- #2158 
### Niche/Secondary Stuff:
- the indication bar at the bottom left corner should have rounded corners to match the browser aesthetics
- Sidebar scrollbar can be made thin to match with the aesthetics of the browser
- Upload Zen browser to other repos #1303 
- Resist tab width fingerprinting #1692

**Comment by d5364t54ytfr4** [comment]
Floorp Browser(another Firefox based browser) has a Private container option. You can right click on any link and you get the Open in  New Tab with Private Container option.

And then there's also `Temporary containers` addon whose developer died and the plugin hasn't been updated since a long time.

Btw Container Sidebar is really useful in normal Firefox.

Also here's another thread about Private Containers:
https://github.com/zen-browser/desktop/discussions/821
